test/main_contracts/rate_oracles/compoundBorrowRateOracle.ts

```diff
@@ -0,0 +1,129 @@
+import { ethers, waffle } from "hardhat";
+import { BigNumber, Wallet } from "ethers";
+import { expect } from "chai";
+import { ConfigForGenericTests as Config } from "./compoundBorrowConfig";
+import { MockCToken, TestCompoundBorrowRateOracle } from "../../../typechain";
+import { advanceTimeAndBlock } from "../../helpers/time";
+
+const { provider } = waffle;
+
+describe("Compound Borrow Rate Oracle", () => {
+  const blocksPerYear = BigNumber.from(31536000).div(13);
+  const wad = BigNumber.from(10).pow(18);
+  const halfWad = BigNumber.from(10).pow(9);
```

Comment:
hmm,  is halfWad not equal to BigNumber.from(5).pow(17)? wdyt @0xZenus ?

Reply:
I think it's neither of these things, but is `BigNumber.from(10).pow(17).mul(5)` 😄

Reply:
naming problem... it's supposed to be 1e9. Is `semiWad` more representative of this?

Reply:
oh yeah, ofc lol

Reply:
oh i see, yeah semiWad is fine:)